lib/node_modules/@stdlib/iter/while-each/README.md

```diff
@@ -0,0 +1,243 @@
+<!--
+
+@license Apache-2.0
+
+Copyright (c) 2023 The Stdlib Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-->
+
+# whileEach Iterator
+
+> Create an iterator which, while a test condition is true, invokes a function for each iterated value before returning the iterated value.
+
+<!-- Section to include introductory text. Make sure to keep an empty line after the intro `section` element and another before the `/section` close. -->
+
+<section class="intro">
+
+</section>
+
+<!-- /.intro -->
+
+<!-- Package usage documentation. -->
+
+<section class="usage">
+
+## Usage
+
+```javascript
+var iterWhileEach = require( '@stdlib/iter/while-each' );
+```
+
+#### iterWhileEach( iterator, predicate, fcn\[, thisArg] )
+
+Returns an iterator which invokes a `function` for each iterated value untill either a `predicate` returns `false` or the function has iterated over all value **before** returning the iterated value.
+
+```javascript
+var array2iterator = require( '@stdlib/array/to-iterator' );
+
+function predicate( v ) {
+    return v < 3;
+}
+
+function assert( v ) {
+    if ( v !== v ) {
+        throw new Error( 'should not be NaN' );
+    }
+}
+
+var it = iterWhileEach( array2iterator( [ 1, 2, 3, 4 ] ), predicate, assert );
+// returns {}
+
+var r = it.next().value;
+// returns 1
+
+r = it.next().value;
+// returns 2
+
+r = it.next().value;
+// undefined
+
+// ...
+```
+
+The returned iterator protocol-compliant object has the following properties:
+
+-   **next**: function which returns an iterator protocol-compliant object containing the next iterated value (if one exists) assigned to a `value` property and a `done` property having a `boolean` value indicating whether the iterator is finished.
+-   **return**: function which closes an iterator and returns a single (optional) argument in an iterator protocol-compliant object.
+
+The invoked `function` is provided two arguments:
+
+-   `value`: iterated value
+-   `index`: iteration index (zero-based)
+
+```javascript
+var array2iterator = require( '@stdlib/array/to-iterator' );
+
+function predicate( v ) {
+    return v < 3;
+}
+
+function assert( v ) {
+    if ( v !== v ) {
+        throw new Error( 'should not be NaN' );
+    }
+}
```

Comment:
```suggestion
function assert( v, i ) {
    if ( i > 1 ) {
        throw new Error( 'unexpected error' );
    }
}
```

I see that `iter/for-each` has a similar example, which does not actually illustrate anything new. Here, we at least want to show that provided functions are invoked with multiple arguments.

Reply:
Once this PR is merged, feel free to submit a PR against `iter/for-each` which updates the respective example to be more illustrative.